pkg/awsvpc/parser.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/logging"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+type AWSresourcesContainer struct {
+	aws.ResourcesContainer
+}
+
+// parseResourcesFromFile returns aws.ResourcesContainer object, containing the configured resources structs
+// from the input JSON file
+func (rc *AWSresourcesContainer) ParseResourcesFromFile(fileName string) error {
+	inputConfigContent, err := os.ReadFile(fileName)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(inputConfigContent, &rc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func mergeResourcesContainers(rc1, rc2 *AWSresourcesContainer) (*AWSresourcesContainer, error) {
+	if rc2 == nil && rc1 != nil {
+		return rc1, nil
+	}
+	if rc2 != nil && rc1 == nil {
+		return rc2, nil
+	}
+	if rc2 == nil && rc1 == nil {
+		return nil, fmt.Errorf("error merging input vpc resources files")
+	}
+	rc1.VpcsList = append(rc1.VpcsList, rc2.VpcsList...)
+	rc1.SubnetsList = append(rc1.SubnetsList, rc2.SubnetsList...)
+	rc1.NetworkACLsList = append(rc1.NetworkACLsList, rc2.NetworkACLsList...)
+	rc1.SecurityGroupsList = append(rc1.SecurityGroupsList, rc2.SecurityGroupsList...)
+	rc1.InternetGWList = append(rc1.InternetGWList, rc2.InternetGWList...)
+	rc1.InstancesList = append(rc1.InstancesList, rc2.InstancesList...)
+
+	return rc1, nil
+}
+
+func (rc *AWSresourcesContainer) VpcConfigsFromFiles(fileNames []string, vpcID, resourceGroup string, regions []string) (
+	*vpcmodel.MultipleVPCConfigs, error) {
+	for _, file := range fileNames {
+		mergedRC := &AWSresourcesContainer{}
+		err1 := mergedRC.ParseResourcesFromFile(file)
+		if err1 != nil {
+			return nil, fmt.Errorf("error parsing input vpc resources file: %w", err1)
+		}
+		rc, err1 = mergeResourcesContainers(mergedRC, rc)
+		if err1 != nil {
+			return nil, err1
+		}
+	}
+	vpcConfigs, err2 := rc.VPCConfigsFromResources(vpcID, resourceGroup, regions)
+	if err2 != nil {
+		return nil, fmt.Errorf("error generating cloud config from input vpc resources file: %w", err2)
+	}
+	return vpcConfigs, nil
+}
+
+func (rc *AWSresourcesContainer) filterByVpc(vpcID string) map[string]bool {
+	shouldSkipVpcIds := make(map[string]bool)
+	for _, vpc := range rc.VpcsList {
+		if vpcID != "" && *vpc.VpcId != vpcID {
+			shouldSkipVpcIds[*vpc.VpcId] = true
+		}
+	}
+	return shouldSkipVpcIds
+}
+
+// VPCConfigsFromResources returns a map from VPC UID (string) to its corresponding VPCConfig object,
+// containing the parsed resources in the relevant model objects
+func (rc *AWSresourcesContainer) VPCConfigsFromResources(vpcID, resourceGroup string, regions []string) (
+	*vpcmodel.MultipleVPCConfigs, error) {
+	res := vpcmodel.NewMultipleVPCConfigs("AWS Cloud") // map from VPC UID to its config
+	var err error
+
+	// map to filter resources, if certain VPC, resource-group or region list to analyze is specified,
+	// skip resources configured outside that VPC
```

Comment:
Not clear enough

Reply:
`// map to filter resources, if certain VPC to analyze is specified,
	// skip resources configured outside that VPC`

updated

Reply:
later we can maybe also support filtering by region and resourceGroup